cmds/identityd/main.go

```diff
@@ -172,3 +180,56 @@ func getIdentityMgr(root string, debug bool) (pkg.IdentityManager, error) {
 
 	return manager, nil
 }
+
+func manageSSHKeys(ctx context.Context) error {
+	authorizedKeysPath := filepath.Join("/", "root", ".ssh", "authorized_keys")
+	err := os.Remove(authorizedKeysPath)
+	if err != nil {
+		return err
+	}
+
+	env := environment.MustGet()
+	runningMode := env.RunningMode
+
+	if runningMode == environment.RunningMain {
+		if env.FarmID != 1 {
+			return nil
+		}
+		runningMode = environment.RunningTest
+	}
+
+	config, err := environment.GetConfigForMode(runningMode)
+	if err != nil {
+		return err
+	}
+
+	authorizedUsers := config.Users.Authorized
+	extraUser, found := kernel.GetParams().GetOne("authorized_user")
+
+	if found {
+		authorizedUsers = append(authorizedUsers, extraUser)
+	}
+
+	github := github.NewClient(nil)
+	var allKeys string
+
+	for _, user := range authorizedUsers {
+		if err := retry.Do(ctx, retry.WithMaxRetries(3, retry.NewConstant(1*time.Microsecond)), func(ctx context.Context) error {
+			keys, _, err := github.Users.ListKeys(ctx, user, nil)
```

Comment:
Please use normal http.Get as i explained above. Then you can do `io.Copy` of the body of the response directly to the open file of authorized_keys. 

It's also important to handle the status code of the request correctly. Some error codes (say 404) means u shouldn't retry and can assume it's a permanent error.